test/test_query.py

```diff
@@ -714,6 +714,206 @@ def test_match_after_set_none(self):
         self.assertInResult(item, matched)
 
 
+class NotQueryMatchTest(_common.TestCase):
+    """Test `query.NotQuery` matching against a single item, using the same
+    cases and assertions as on `MatchTest`, plus assertion on the negated
+    queries (ie. assertTrue(q) -> assertFalse(NotQuery(q))).
+    """
+    def setUp(self):
+        super(NotQueryMatchTest, self).setUp()
+        self.item = _common.item()
+
+    def test_regex_match_positive(self):
+        q = dbcore.query.RegexpQuery('album', '^the album$')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_regex_match_negative(self):
+        q = dbcore.query.RegexpQuery('album', '^album$')
+        self.assertFalse(q.match(self.item))
+        self.assertTrue(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_regex_match_non_string_value(self):
+        q = dbcore.query.RegexpQuery('disc', '^6$')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_substring_match_positive(self):
+        q = dbcore.query.SubstringQuery('album', 'album')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_substring_match_negative(self):
+        q = dbcore.query.SubstringQuery('album', 'ablum')
+        self.assertFalse(q.match(self.item))
+        self.assertTrue(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_substring_match_non_string_value(self):
+        q = dbcore.query.SubstringQuery('disc', '6')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_year_match_positive(self):
+        q = dbcore.query.NumericQuery('year', '1')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_year_match_negative(self):
+        q = dbcore.query.NumericQuery('year', '10')
+        self.assertFalse(q.match(self.item))
+        self.assertTrue(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_bitrate_range_positive(self):
+        q = dbcore.query.NumericQuery('bitrate', '100000..200000')
+        self.assertTrue(q.match(self.item))
+        self.assertFalse(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_bitrate_range_negative(self):
+        q = dbcore.query.NumericQuery('bitrate', '200000..300000')
+        self.assertFalse(q.match(self.item))
+        self.assertTrue(dbcore.query.NotQuery((q,)).match(self.item))
+
+    def test_open_range(self):
+        q = dbcore.query.NumericQuery('bitrate', '100000..')
+        dbcore.query.NotQuery((q,))
+
+
+class NotQueryTest(DummyDataTestCase):
```

Comment:
These tests might need some tweaking or adding some fields to the `DummyData` set - at the moment, more tests than I'd like are matching empty sets, are not too varied, etc.